src/server/services/adoptable-trash-containers.ts

```diff
@@ -0,0 +1,164 @@
+    (result) => {
+      return {
+        ...result,
+        id: datasetId,
+      };
+    }
+  );
+  const MY_LOCATION = await fetchMyLocation(requestID, authProfileAndToken);
+
+  if (
+    MY_LOCATION.status !== 'OK' ||
+    !MY_LOCATION.content?.[0]?.latlng ||
+    meldingen?.status !== 'OK'
+  ) {
+    return apiDependencyError({ MY_LOCATION });
+  }
+
+  sortLatLngCoordinates(meldingen.content.features);
+
+  const latlng = MY_LOCATION.content[0].latlng;
+  const WITHIN_RADIUS_KM = 0.1;
+  const featuresInRadius = filterFeaturesinRadius(
+    latlng,
+    meldingen.content.features,
+    WITHIN_RADIUS_KM
+  );
+
+  const filteredFeatures: MaPointFeature<AfvalFeatureProperties>[] =
+    filterDatasetFeatures<AfvalFeatureProperties>(
+      featuresInRadius,
+      [datasetId],
+      filters
+    );
+
+  return apiSuccessResult({
+    tips: buildNotifications(
+      filteredFeatures,
+      await fetchBRP(requestID, authProfileAndToken)
+    ),
+  });
+}
+
+/** Put all LatLngCoordinates in the right order. */
+function sortLatLngCoordinates(
+  features: DatasetFeatures<DatasetFeatureProperties>
+): void {
+  for (const feature of features) {
+    if (feature.geometry.type === 'Point') {
+      const { lat, lng } = getLatLngCoordinates(
+        feature.geometry.coordinates as LatLngTuple
+      );
+      feature.geometry.coordinates = [lat, lng];
+    } else {
+      captureMessage(`Unexpected geometry type: '${feature.geometry.type}'`, {
+        severity: 'error',
+      });
+    }
+  }
+}
+
+type AfvalFeatureProperties = DatasetFeatureProperties & {
+  geadopteerd_ind: 'Ja' | 'Nee';
+};
+
+/** Build notifications.
+ *
+ * Only returns the last notification to prevent spam.
+ */
+function buildNotifications(
+  features: MaPointFeature<AfvalFeatureProperties>[],
+  brpResponse: ApiResponse<BRPData>
+): MyNotification[] {
+  if (
+    !FeatureToggle.adopteerbareAfvalContainerMeldingen ||
+    !features.length ||
+    brpResponse.status !== 'OK' ||
+    !brpResponse.content.persoon.geboortedatum
+  ) {
+    return [];
+  }
+
+  const birthday = new Date(brpResponse.content.persoon.geboortedatum);
+
+  const descriptionText = determineDescriptionText(birthday);
+  if (!descriptionText) {
+    return [];
+  }
+
+  const [lat, lng] = features[0].geometry.coordinates;
+  return [buildNotification(descriptionText, lat, lng)];
+}
+
+function determineDescriptionText(birthday: Date): string | undefined {
+  const age = differenceInYears(new Date(), birthday);
+  const ADULT_AGE = 18;
+  const LATE_TEEN_AGE = 16;
+
+  if (age >= ADULT_AGE) {
+    return `Help mee om uw eigen buurt schoon te houden en adopteer een afvalcontainer.
+ Liever op een andere manier bijdragen? Leen dan een afvalgrijper!`;
+  } else if (age >= LATE_TEEN_AGE) {
+    return `Help mee om je eigen buurt schoon te houden en adopteer een afvalcontainer.
+ Wil je liever iets anders doen? Leen dan een afvalgrijper!`;
+  }
+  return undefined;
```

Comment:
Wanneer je al eerder checked (zie mijn eerdere opmerkingen) of je hier überhaupt terecht moet komen (`age < LATE_TEEN_AGE`) kun je deze code cleaner maken.

```suggestion
  if (age < ADULT_AGE) {
    return `Help mee om je eigen buurt schoon te houden en adopteer een afvalcontainer.
 Wil je liever iets anders doen? Leen dan een afvalgrijper!`;
  }
  
  // All other
  return `Help mee om uw eigen buurt schoon te houden en adopteer een afvalcontainer.
 Liever op een andere manier bijdragen? Leen dan een afvalgrijper!`;

```